### PR TITLE
Refactoring code around test harness

### PR DESCRIPTION
**Pull request by mjp41**
Small changes to make the test harness write to a log file in dump mode, and tidy up some of the error message production.

This is a precursor to larger test harness refactoring.

### CODE REVIEW

File: src/interpreter/value.cc [hunk 1]
```diff
@@ -97,8 +98,8 @@ namespace verona::interpreter
   {
     if (tag != UNINIT)
     {
-      std::cerr << "Dropped an initialized Value" << std::endl;
-      abort();
+      // TODO: Should this be an internal error?
```
Comment (plietar) [inline src/interpreter/value.cc:101]: I think there is no way for bad bytecode to trigger this, unless it’s a VM bug, so internal error seems fine to me.

There’s actually an explicit check in the return opcode to make sure all registers in the frame have already been cleared, in which case it calls `fatal`.

File: src/compiler/error.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#pragma once
```
Comment (plietar) [inline src/compiler/error.h:1]: Since this is used by the VM, should this live somewhere else? eg. `src/ds/`

**Review by plietar — APPROVED** [review]

File: src/compiler/error.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <iostream>
+#include <type_traits>
+#include <fmt/ostream.h>
+
+
+class InternalError
```
[the remark this answers is not in the document]
Comment (mjp41) [reply src/compiler/error.h:8]: Yeah, was trying to minimise the diff, can consolidate.

File: src/compiler/error.h (hunk 2 above)
Comment (mjp41) [reply src/compiler/error.h:1]: Good point

File: src/compiler/error.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (mjp41) [reply src/compiler/error.h]: Did this in the end as MSVC gets upset about non-returning destructors.
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4722?view=msvc-160

**Pull request merged**